### PR TITLE
Add remote doctypes for app LaMusiqueDeMesFilms .

### PR DESCRIPTION
**Pull request by jacquarg**
I didn't manage to test it, and I don't expect that it will works without testing yet.

I saw the [for developers](https://cozy.github.io/cozy-stack/remote.html#for-developers) paragraph in the docs, but I can't pass new parameters easily with cozy-app-dev docker image (even with docker consle, image quit as soon as I stop cozy-stack quit, so I can restart it with additionnal parameters ...)

Have-you any advice to test it ?

ping @poupotte

### CODE REVIEW

File: com.deezer.api.track/request [hunk 1]
```diff
@@ -0,0 +1,2 @@
+GET https://api.deezer.com/search/track/?output=json&strict=on&q={{q}}
+  Accept "application/json"
```
Comment (nono) [inline com.deezer.api.track/request]: For the Accept header, you should remove the spaces at the beginning of the line and use this format:

```
Accept: application/json
```

File: mirror.musicbrainz.recording.search/request [hunk 2]
```diff
@@ -0,0 +1,2 @@
+GET https://ssl14.ovh.net/~hoodbrai/proxy.php?http://musicbrainz-mirror.eu:5000/ws/2/recording?fmt=json&query={{q}}
```
Comment (nono) [inline mirror.musicbrainz.recording.search/request]: Using a proxy is not mandatory. We don't enforce https between the cozy-stack and the remote service.

Comment (jacquarg) [inline mirror.musicbrainz.recording.search/request]: Great ! thanks !

**Comment by nono** [comment]
> Have-you any advice to test it ?

Sorry, excepting installing it from source (and using docker only for couchdb), I don't see a way to test it for now. I'll try to see if I can find something for that soon, but I'm not very inspired for now.

**Comment by jacquarg** [comment]
Another question:
What is the principle for images ? 
For json data, I use cozy.client.fetchJSON ; to easily add Bearer header. 

But for images, I can't add a bearer header to an img tag.

Should I fetch the image data with ajax, and then populate the img ; or is there a easier way ?

**Comment by nono** [comment]
For the images, it's a bit complicated. You can use the fetch API to download them, and populate the img. But we can use the CSP whitelist for them now.

Later, we will add something easier for proxying images. Probably with a generated secret on the URL to avoid having to send the token as a bearer header.

**Comment by jacquarg** [comment]
To use the cozy-app-dev docker image ; if it was possible to add the doctypes directory in the docker command, as it is done with the storage. ie  `-v mydif/cozy-doctypes/:/data/cozy-doctypes` ; it would be nice : )

**Comment by jacquarg** [comment]
Ok for the images ; I will use the fetch API for this app ; as I may needs various sources in the near future.

But I remember the whitelist option, for other apps (and tiles !). I don't know yet how versatile a library like leaflet will be, about tiles request customization.

Thanks !

**Comment by jacquarg** [comment]
@nono, @poupotte ?
I have a stack up and running on my desktop, everything looks OK ; but the --doctypes option doesn't seems to work. Only the remote doctypes already defined (ie org.wikidata.search) work.
This is my serve command : 
`./cozy-stack serve --doctypes '/home/jacquarg/workspace/cozy_v3/cozy-doctypes/' --appdir ~/workspace/orange-musiquedemesfilms/`

**Comment by nono** [comment]
@jacquarg what do you have in the logs? What do you have in the `/home/jacquarg/workspace/cozy_v3/cozy-doctypes/` directory? What does `cozy-stack config print` output?

**Comment by jacquarg** [comment]
Thanks for quick reply !

logs said 403 forbidden for my tries, 
```
	ERRO[0098] [http] GET /remote/org.wikidata.searchbis code=403, message=Forbidden  domain="cozy.tools:8080"
time=2017-06-14T10:26:12+02:00	status=403	method=GET	host=cozy.tools:8080	uri=/remote/org.wikidata.searchbis?q=batman	bytes_out=9
```

but ok for the one in the master of cozy-doctypes :
```
time=2017-06-14T10:27:09+02:00	status=200	method=GET	host=cozy.tools:8080	uri=/remote/org.wikidata.search?q=batman	bytes_out=2189
```


I clone cozy-doctypes repository, and add some doctypes : 
```
ls /home/jacquarg/workspace/cozy_v3/cozy-doctypes/

com.deezer.api.track                 mirror.musicbrainz.release-group         org.wikidata.sparql
docs                                 mirror.musicbrainz.release-group.search  org.wikidata.wbsearchentities
LICENSE                              org.wikidata.entity                      org.wikipedia.en.api
mirror.musicbrainz.recording.search  org.wikidata.search                      org.wikipedia.fr.api
mirror.musicbrainz.release           org.wikidata.searchbis
```


*cozy-stack config print*

```
./cozy-stack config print
{
  "Host": "localhost",
  "Port": 8080,
  "Assets": "",
  "Doctypes": "",
  "Subdomains": "nested",
  "AdminHost": "localhost",
  "AdminPort": 6060,
  "NoReply": "",
  "Fs": {
    "Auth": null,
    "URL": {
      "Scheme": "file",
      "Opaque": "",
      "User": null,
      "Host": "localhost",
      "Path": "/home/jacquarg/workspace/cozy_v3/cozy-stack/storage",
      "RawPath": "",
      "ForceQuery": false,
      "RawQuery": "",
      "Fragment": ""
    }
  },
  "CouchDB": {
    "Auth": null,
    "URL": {
      "Scheme": "http",
      "Opaque": "",
      "User": null,
      "Host": "localhost:5984",
      "Path": "/",
      "RawPath": "",
      "ForceQuery": false,
      "RawQuery": "",
      "Fragment": ""
    }
  },
  "Jobs": {
    "Workers": 4,
    "Redis": {
      "Auth": null,
      "URL": null
    }
  },
  "Konnectors": {
    "Cmd": ""
  },
  "Mail": {
    "host": "localhost",
    "port": 465,
    "username": "",
    "password": "",
    "native_tls": false,
    "disable_tls": false,
    "SkipCertificateValidation": false,
    "local_name": ""
  },
  "Cache": {
    "Auth": null,
    "URL": null
  },
  "Lock": {
    "Auth": null,
    "URL": null
  },
  "SessionStorage": {
    "Auth": null,
    "URL": null
  },
  "DownloadStorage": {
    "Auth": null,
    "URL": null
  },
  "KonnectorsOauthStateStorage": {
    "Auth": null,
    "URL": null
  }
}
```

**Comment by nono** [comment]
403, it looks a permission issue. Do you have a permission declared for `org.wikidata.searchbis` in the manifest of your app? When using `--appdir`, the permissions for the manifest are read when loading the index of a route: you have to reload the page in the browser, not just edit the manifest for the change to be effective.

**Comment by jacquarg** [comment]
oOps ; sorry, my bad, I just completely forgot about the declaration of remotes doctypes in the manifest.

It's working \o/

**Comment by jacquarg** [comment]
It's almost ok, but I have a problem with the wikidata sparql endpoint :
I got a 502 : the content-type for the response is not authorized
The content type is : application/sparql-results+json


And some helpers in cozy-client-js will be helpfull in the future, as some content may need encodeURIComponent (twice sometimes) ; and I am not sure that the content-type of the proxyed response is well transmitted (I have to JSON.parse the result).

I start working on pictures !

**Comment by nono** [comment]
OK, for the sparql content-type, we will allow it. We only allow some content-type, because I'd like to avoid things like javascript or HTML and I'm not really sure about other content-types. So we have started with a small list of allowed content-types, and we will see how it grows.

-> https://github.com/cozy/cozy-stack/pull/719

For the helpers, I take notes. We will try to do it in the coming weeks.

**Comment by nono** [comment]
@jacquarg do you want to make other changes? Or should we merge?

**Comment by jacquarg** [comment]
I just test with the updated stack. And things seems to work ! (fetching images with ajax is tricky !)

I'm ready fir merge : )

**Pull request merged**